### ISSUE TITLE
@keystone-6/core reports a vulnerable version of undici

### ISSUE THREAD

**Issue by taylorsweetman** [keystonejs/keystone#8356]
When installing the `@keystone-6/core` module with `npm`, there are 5 high severity vulnerabilities in dependencies flagged by `npm audit`. Seems the solution is to bump the version of the `@prisma/engine-core` dependency. 

A very similar output from `yarn audit` happens when building the project with `yarn` instead of `npm`. 

## Steps to reproduce
1. Initialize npm in a new project, `npm init`
2. Install the latest version of core, `npm i @keystone-6/core`
3. Run an audit, `npm audit`

## Expected behaviour

No vulnerabilities found from `npm audit`, especially on relatively new packages. 

## NPM Audit Output
``` # npm audit report

undici  <=5.19.0
Severity: high
Regular Expression Denial of Service in Headers - https://github.com/advisories/GHSA-r6ch-mqf9-qc9w
CRLF Injection in Nodejs ‘undici’ via host - https://github.com/advisories/GHSA-5r9g-qh6m-jxff
fix available via `npm audit fix --force`
Will install @keystone-6/core@2.3.1, which is a breaking change
node_modules/undici
  @prisma/engine-core  2.0.0-1 - 2.0.0-previewnull-1 || 2.0.1-1 - 2.0.1-dev.3 || >=2.1.0-dev.1
  Depends on vulnerable versions of undici
  node_modules/@prisma/engine-core
    @prisma/internals  *
    Depends on vulnerable versions of @prisma/engine-core
    node_modules/@prisma/internals
      @keystone-6/core  <=0.0.0-rc-20230220024700 || >=3.0.0
      Depends on vulnerable versions of @prisma/internals
      Depends on vulnerable versions of @prisma/migrate
      node_modules/@keystone-6/core
      @prisma/migrate  3.0.3-dev.1 - 3.0.3-dev.2 || >=3.16.0-dev.1
      Depends on vulnerable versions of @prisma/internals
      node_modules/@prisma/migrate

5 high severity vulnerabilities
```

## Context
- Node version: 18.14.2
- npm version: 9.5.0

## Friendly suggestion
Would it make sense to include an `npm audit` and/or `yarn audit` check in the CI tooling?

**Comment by borisno2** [comment]
Hi @taylorsweetman 
Thanks for logging this issue. A PR to upgrade Prisma to 4.11.0 (#7964) was merged yesterday and will be released as a new Keystone minor version soon.

**Comment by taylorsweetman** [comment]
Great news! Appreciate the speedy reply, @borisno2.

**Comment by borisno2** [comment]
No worries @taylorsweetman we have just released 5.1 with the latest Prisma version https://github.com/keystonejs/keystone/releases/tag/2023-03-02

**Comment by acburdine** [comment]
@borisno2 I just installed 5.1.0, unfortunately it doesn't look like it fixed either of the `undici` vulnerabilities - `@prisma/engine-core@4.11.0` has its version of undici pinned to 5.16.0 (see [here](https://github.com/prisma/prisma/blob/4.11.0/packages/engine-core/package.json#L47)).

Doesn't look like Keystone can really do anything at the moment until Prisma updates its dependency version, but it might be worth updating the release notes?

**Comment by dcousens** [comment]
Thanks @acburdine

**Comment by dcousens** [comment]
Update from https://github.com/prisma/prisma/pull/17964 is that this vulnerability is a vulnerability in name only, and should not be possible to exploit.
I'll keep the issue open insofar as vulnerability scanners will prompt users to find this issue until it is resolved.

**Comment by dcousens** [comment]
https://github.com/prisma/prisma/pull/17964 has been merged upstream!
We'll update as soon as it's available.

**Comment by acburdine** [comment]
@dcousens I believe this issue can be closed now - `@keystone-6/core@5.2.0` uses prisma 4.13.0  which has an up-to-date version of undici.

**Comment by borisno2** [comment]
Thanks @acburdine
